src/test/integrationTests/requestController.test.js

```diff
@@ -74,24 +75,121 @@ describe('Integration tests for the request controller', () => {
   });
   describe('Validation tests for the book a trip features', () => {
     it('should not book a one way trip when a required detail is missing', async () => {
-      delete tripDetails.flightDate;
+      delete tripDetails.origin;
       const response = await chai.request(app).post('/api/v1/request/book_trip')
         .set('x-access-token', token).send(tripDetails);
       expect(response.status).to.equal(400);
       expect(response.body).to.have.property('message');
-      expect(response.body.message).to.equal('The "flightDate" field is required');
+      expect(response.body.message)
+        .to.equal('The "origin" field is required');
       expect(response.body).to.have.property('success');
       expect(response.body.success).to.equal(false);
     });
     it('should not book a return trip when a required detail is missing', async () => {
-      delete tripDetails.returnDate;
+      delete tripDetails.origin;
       const response = await chai.request(app).post('/api/v1/request/book_return_trip')
         .set('x-access-token', token).send(tripDetails);
       expect(response.status).to.equal(400);
       expect(response.body).to.have.property('message');
-      expect(response.body.message).to.equal('The returnDate field is required.');
+      expect(response.body.message)
+        .to.equal('The "origin" field is required');
       expect(response.body).to.have.property('success');
       expect(response.body.success).to.equal(false);
     });
   });
+  it('should allow a registered user to book a multcity trip', async () => {
+    const response = await chai.request(app).post('/api/v1/request/multicity')
+      .set('x-access-token', token).send({
+        origin: 'Onipan',
+        destination: ['mile12', 'okoko'],
+        flightDate: ['2019-06-27', '2019-06-25'],
```

Comment:
What happens when user enters inconsistent _destination_ and _flightDate_ properties? An example is given below:
```
{
  ...
  destination: ['mile12', 'okoko'],
  flightDate: ['2019-06-27'],
}
```

Did you consider this in your error catching? If not, please do. A good error message could be _'Destination places and flight dates do not match.'_; something like that.

Thanks.

Reply:
i did consider it thanks